src/main/kotlin/sudoklin/solver/SudokuSolver.kt

```diff
@@ -183,4 +184,181 @@ class SudokuSolver(initialSudoku: Sudoku, val log: Boolean = false) {
             }
         }
     }
+}
+
+class NewSudokuSolver(val sudoku: NewShinySudoku, val log: Boolean = false) {
+    val timestamp: String = (System.currentTimeMillis() / (60 * 1000)).toString()
+
+    fun getMissingNumbersForRow(rowIndex: Int): List<Int> {
```

Comment:
can we merge `getMissingNumbersForRow`, `getMissingNumbersForColumn` and `getMissingNumbersForGroup` to reduce duplicated code?